### PR TITLE
Twitter Direct Message

### PR DESCRIPTION
**Pull request by claudio4j**
https://github.com/syndesisio/syndesis/issues/3521

### CODE REVIEW

**Comment by oscerd** [comment]
There are some errors:

`[INFO] --- maven-dependency-plugin:3.1.1:analyze-only (basepom.default) @ connector-twitter ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.apache.camel:camel-core:jar:2.21.0.fuse-740013:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.springframework:spring-context:jar:4.3.19.RELEASE:compile
[WARNING]    org.apache.camel:camel-twitter:jar:2.21.0.fuse-740013:compile`

File: app/connector/twitter/pom.xml [hunk 1]
```diff
@@ -15,70 +15,87 @@
...
-      <groupId>org.apache.camel</groupId>
-      <artifactId>camel-twitter</artifactId>
-      <scope>runtime</scope>
-    </dependency>
-  </dependencies>
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+        </dependency>
```
Comment (oscerd) [inline app/connector/twitter/pom.xml]: I don't think spring-context is needed

File: app/connector/twitter/pom.xml [hunk 2]
```diff
@@ -15,70 +15,87 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+            <scope>provided</scope>
+        </dependency>
```
Comment (oscerd) [inline app/connector/twitter/pom.xml]: All the spring-boot stuff have been removed from others connectors. This is the actual master POM for twitter connector
https://github.com/syndesisio/syndesis/blob/master/app/connector/twitter/pom.xml

Comment (claudio4j) [reply app/connector/twitter/pom.xml]: Fixed.

File: app/connector/twitter/pom.xml (hunk 1 above)
Comment (claudio4j) [reply app/connector/twitter/pom.xml]: Fixed.

**Comment by zregvart** [comment]
@claudio4j only few bits with the [license headers](https://circleci.com/gh/syndesisio/syndesis/72626) left:
```
[WARNING] Missing header in: /workspace/app/connector/twitter/src/main/java/io/syndesis/connector/twitter/DirectMessageText.java
[WARNING] Missing header in: /workspace/app/connector/twitter/src/main/java/io/syndesis/connector/twitter/SendDirectMessageCustomizer.java
```

You can do a `./mvnw -N license:format` from `app` directory to fix those.

**Comment by claudio4j** [comment]
> @claudio4j only few bits with the [license headers](https://circleci.com/gh/syndesisio/syndesis/72626) left:

The header was there, but formatted slightly different. Thank you.

**Comment by zregvart** [comment]
Seems that we have some dependency issues left:
```
[INFO] --- maven-dependency-plugin:3.1.1:analyze-only (basepom.default) @ connector-twitter ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.apache.camel:camel-core:jar:2.21.0.fuse-740013:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.apache.camel:camel-twitter:jar:2.21.0.fuse-740013:compile
```

I think declaring `camel-core` in `compile` and `camel-twitter` in `runtime` scope should fix it.

You can run the checks locally with `syndesis build -m connector` or just the Twitter connector with `./mvnw verify -pl :connector-twitter -am`.

**Comment by claudio4j** [comment]
> Seems that we have some dependency issues left:

I think it is fine now. Thanks for the review @zregvart.

**Comment by zregvart** [comment]
CI build fails with:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project connector-twitter: Compilation failure
[ERROR] /workspace/app/connector/twitter/src/main/java/io/syndesis/connector/twitter/SendDirectMessageCustomizer.java:[22,24] package org.apache.camel does not exist
```

I think `camel-core` needs to be in `compile` scope.

**Comment by zregvart** [comment]
:crying_cat_face: there's a test that assumes something about the Twitter connector in the server, so now a test fails there:

```
[ERROR] Tests run: 12, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.601 s <<< FAILURE! - in io.syndesis.server.dao.DataManagerTest
[ERROR] getTwitterConnector(io.syndesis.server.dao.DataManagerTest)  Time elapsed: 0.165 s  <<< FAILURE!
java.lang.AssertionError: expected:<2> but was:<4>
	at io.syndesis.server.dao.DataManagerTest.getTwitterConnector(DataManagerTest.java:106)
```

The UI build is failing hopefully #5476 will fix it and then you can rebase on top of it.

**Comment by claudio4j** [comment]
> there's a test that assumes something about the Twitter connector in the server, so now a test fails there:

previously I built with -f option, so no tests, lesson learned before a PR. Hopefully the problems are gone now. Thanks.

**Pull request merged**